### PR TITLE
HL-281 test on CI

### PR DESCRIPTION
**Pull request by pondpiu**
[Link to ticket](https://dltx.atlassian.net/jira/software/projects/HL/boards/82?selectedIssue=HL-281)

- GH action to run unit test on CI
- add typing to dataProviders

### CODE REVIEW

**Review by OptimusOpus — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -7,15 +7,17 @@
...
     "dev": "yarn codegen && nodemon ./src/app.ts",
     "build": "yarn codegen && tsc",
     "start": "node build/app.js",
     "format": "prettier -w src",
     "lint": "pretty-quick --staged",
     "typechain": "typechain --target ethers-v5 ./src/abis/*.json --out-dir ./src/typechain",
     "codegen": "yarn typechain && tsoa spec-and-routes",
-    "mocha-test": "mocha -r ts-node/register src/**/*.spec.ts --require src/test/fixtures.ts --timeout 10000"
+    "test:e2e": "mocha -r ts-node/register src/test/e2e/*.spec.ts --require src/test/fixtures.ts --timeout 10000",
+    "test:unit": "mocha -r ts-node/register src/test/unit/*.spec.ts",
```
Comment (brettvniekerk) [inline package.json:19]: should these not run `codegen` beforehand too?

File: src/common/cache.ts [hunk 2]
```diff
@@ -26,9 +28,9 @@ export const getCache = async (key: string) => {
   return await nodeCache.get(key);
```
Comment (brettvniekerk) [inline src/common/cache.ts:28]: dont need `await` with async return statement

File: src/common/cache.ts [hunk 3]
```diff
@@ -26,9 +28,9 @@ export const getCache = async (key: string) => {
...
 };
 
-export const setCache = async (
+export const setCache = async <T = any>(
```
Comment (brettvniekerk) [inline src/common/cache.ts:31]: why not just `<T>`?

File: src/providers/data/runnerDataProvider.ts [hunk 4]
```diff
@@ -2,14 +2,17 @@ import { getCache, setCache } from "../../common/cache";
...
 import { getToday } from "../../common/utils/getToday";
+import { AsyncReturnType } from "../../common/utils/types";
+
+type Runners = AsyncReturnType<typeof getRaceData>;
 
 export const getTodayRaceData = async (track: string, race: string) => {
   const today = getToday("YYYY-MM-DD");
   const marketId = makeMarketId(new Date(), track, race);
-  const cache = await getCache(`race-${marketId}`);
+  const cache = await getCache<Runners>(`race-${marketId}`);
```
Comment (brettvniekerk) [inline src/providers/data/runnerDataProvider.ts:12]: `marketId.toLowerCase()` for safety pls

File: src/providers/data/vaultDataProvider.ts [hunk 5]
```diff
@@ -42,14 +44,15 @@ const getVaultTokenFromChain = async (vaultAddress: string) => {
   };
 };
 
+type VaultToken = AsyncReturnType<typeof getVaultTokenFromChain>;
 export const getVaultToken = async (vaultAddress: string) => {
   const vaultCacheId = vaultAddress.toLowerCase();
-  const cache = await getCache(`vault-${vaultCacheId}-token`);
+  const cache = await getCache<VaultToken>(`vault-${vaultCacheId}-token`);
```
Comment (brettvniekerk) [inline src/providers/data/vaultDataProvider.ts:50]: `vaultCacheId.toLowerCase()` for safety pls

**Review by brettvniekerk — COMMENTED** [review]
Few minor things

File: package.json (hunk 1 above)
Comment (pondpiu) [reply package.json:19]: i feel like its more ergonomic this way as i intend to use unit test to do TDD

**Review by brettvniekerk — DISMISSED** [review]
LGTM - does everything still work with the `makeMarketId` change?

**Review by brettvniekerk — DISMISSED** [review]
Pls test placing and settling bets before merging

**Review by OptimusOpus — APPROVED** [review]

**Review by bitcoinbrisbane — CHANGES_REQUESTED** [review]
Need to add REGISTERY_CONTRACT TO .env.development

**Review by OptimusOpus — APPROVED** [review]

**Review by brettvniekerk — APPROVED** [review]

**Pull request merged**